datafusion/core/src/execution/context.rs

```diff
@@ -931,6 +931,11 @@ impl SessionContext {
     pub fn task_ctx(&self) -> Arc<TaskContext> {
         Arc::new(TaskContext::from(self))
     }
+
+    /// Get a copy of the [`SessionState`] of this [`SessionContext`]
```

Comment:
I'm not a massive fan of this, but it appears to be what we do already for every query and when constructing a TaskContext, so it can't be that bad..........